pkg/bicep/schematobicep.go

```diff
@@ -61,3 +64,196 @@ func getBicepType(schemaType string) (string, error) {
 		return "", errors.New("unknown type: " + schemaType)
 	}
 }
+
+func declareParameter(name string, sch *schema.Schema, buf *bytes.Buffer) error {
+	bicepType, err := getBicepType(sch.Type)
+	if err != nil {
+		return err
+	}
+
+	if sch.Default != nil {
+		err = declareDefault(name, sch, buf)
+		if err != nil {
+			return err
+		}
+		return nil
+	} else {
+		buf.WriteString(fmt.Sprintf("param %s %s\n", name, bicepType))
+	}
+
+	return nil
+}
+
+func declareAllowed(sch *schema.Schema, buf *bytes.Buffer) error {
+	if sch.Enum != nil && len(sch.Enum) > 0 {
+		bicepType, err := getBicepType(sch.Type)
+		if err != nil {
+			return err
+		}
+
+		enumBytes, err := json.MarshalIndent(sch.Enum, "", "    ")
+		if err != nil {
+			return err
+		}
+
+		enumString := string(enumBytes)
+		r := strings.NewReplacer(`"`, `'`, ",", "")
+		cleanString := r.Replace(enumString)
+
+		if bicepType == "object" {
+			declareAllowedObject(cleanString, sch, buf)
+		} else {
+			buf.WriteString(fmt.Sprintf("@allowed(%v)\n", cleanString))
+		}
+	}
+
+	return nil
+}
+
+func declareAllowedObject(str string, sch *schema.Schema, buf *bytes.Buffer) {
```

Comment:
How is this different than `declareDefaultObject`? If both of them render a bicep object then you should consolidate.